lightning/src/ln/monitor_tests.rs

```diff
@@ -398,7 +398,7 @@ fn do_test_claim_value_force_close(prev_commitment_tx: bool) {
 
 	// Broadcast the closing transaction (which has both pending HTLCs in it) and get B's
```

Comment:
@wpaulino Pinging you as you authored a3b416a3 and the `test_restored_packages_retry`. 

For its `true` variant, this test is failing with new config CSV timeout of 7 days of blocks:
```
thread 'ln::monitor_tests::test_restored_packages_retry' panicked at 'assertion failed: `(left == right)`
  left: `OutPoint { txid: 0dfec8a032dd7df936c62f8a43c40ed92b15fdc76a513f955be589092dd4cf45, vout: 0 }`,
 right: `OutPoint { txid: d738d83463d1edb85a557328e0789b1f63807e96d2a3516747f75a0478fd4236, vout: 0 }`', lightning/src/ln/monitor_tests.rs:1719:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

My understanding of the root of the failure is that `serialized_monitor` is hardcoding the old value of 1 day of block as a CSV timeout, therefore when it’s compared to “witness” HTLC timeout transactions, the scriptPubkey has been altered and the previous outpoint differs. So I think I’ve to do a new dump of the monitor state ? However doing so I might break the compatibility check that the `true` variant sounds to verify ?

Reply:
Latest push as Wilmer suggestion to update the old chain monitor deserialized and that packages/transactions are rebroadcast accordingly. However I got a flappy test sometimes works well, and sometimes I got a connection logic error.

```
thread 'ln::monitor_tests::test_restored_packages_retry' panicked at 'assertion failed: `(left == right)`
  left: `7d75aecbfa7227b3911f8dd40c6be8dfce35ae1635551d418d7b65eaf60e60f1`,
 right: `077b978d9bcba00814c8e6d68a4d3290a8f29813c40eb82889c7d4908cca5294`: Transaction d738d83463d1edb85a557328e0789b1f63807e96d2a3516747f75a0478fd4236 was already confirmed and is being re-confirmed in a different block.
This indicates a severe bug in the transaction connection logic - a reorg should have been processed first!', lightning/src/chain/channelmonitor.rs:3162:25
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Note the PR has been rebased.

Happy if you have a look, I’ll dig more into it.

Reply:
Yes I tested both it didn’t yield a success.